src/array.rs

```diff
@@ -20,7 +20,7 @@ impl<'a, A> BTreeSlice<'a, A> {
     }
 }
 
-impl<'a, A: Clone> BTreeSlice<'a, A> {
+impl<A: Clone> BTreeSlice<'_, A> {
```

Comment:
ab42d67c78505463cf66b7f54a4137ea017e51d4: Are you using rust analyzer? `clippy` didn't show me these lints.